trl/trainer/ppo_trainer.py

```diff
@@ -98,14 +96,16 @@ class PPOTrainer(Trainer):
     _tag_names = ["trl", "ppo"]
 
     @deprecate_kwarg("tokenizer", new_name="processing_class", version="0.15.0", raise_if_both_names=True)
+    @deprecate_kwarg("policy", new_name="model", version="0.15.0", raise_if_both_names=True)
+    @deprecate_kwarg("ref_policy", new_name="ref_model", version="0.15.0", raise_if_both_names=True)
     def __init__(
         self,
         config: PPOConfig,
         processing_class: Optional[
             Union[PreTrainedTokenizerBase, BaseImageProcessor, FeatureExtractionMixin, ProcessorMixin]
         ],
-        policy: nn.Module,
-        ref_policy: Optional[nn.Module],
+        model: nn.Module,
```

Comment:
If I'm not mistaken, won't this breaking change be introduced in the next TRL release (i.e. `v0.13.0` instead of `0.150`)? In other words, don't we technically need to support both sets of args until `v0.15`?

Reply:
The decorator allows for supporting both args, and when the old one is used, raises a warning:

```python
>>> from datasets import load_dataset
>>> from transformers import AutoModelForCausalLM, AutoModelForSequenceClassification, AutoTokenizer
>>> from trl import PPOConfig, PPOTrainer
>>> args = PPOConfig("my_out_dir")
>>> tokenizer = AutoTokenizer.from_pretrained("EleutherAI/pythia-1b-deduped")
>>> model = AutoModelForCausalLM.from_pretrained("EleutherAI/pythia-1b-deduped")
>>> ref_model = AutoModelForCausalLM.from_pretrained("EleutherAI/pythia-1b-deduped")
>>> reward_model = AutoModelForSequenceClassification.from_pretrained("EleutherAI/pythia-1b-deduped", num_labels=1)
>>> value_model = AutoModelForSequenceClassification.from_pretrained("EleutherAI/pythia-1b-deduped", num_labels=1)
>>> dataset = load_dataset("trl-internal-testing/descriptiveness-sentiment-trl-style", split="descriptiveness")
>>> trainer = PPOTrainer(
...      config=args,
...      processing_class=tokenizer,
...      policy=model,  # old arg here
...      ref_model=ref_model,
...      reward_model=reward_model,
...      value_model=value_model,
...      train_dataset=dataset,
...  )
/fsx/qgallouedec/transformers/src/transformers/utils/deprecation.py:165: FutureWarning: `policy` is deprecated and removed starting from version 0.15.0 for `PPOTrainer.__init__`. Use `model` instead.
  return func(*args, **kwargs)
```

Did I misunderstand your point?

Reply:
Ah I was mistaken indeed. This looks great then!